### PR TITLE
GUI_Portfolio page

### PR DESCRIPTION
**Pull request by nmamro**

### CODE REVIEW

File: Clients/GUI/Pages/Portfolio.razor [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+                            </tr>
+                        }
+                    </tbody>
+                </table>
+
+                <div class="btn-group-sm">
+                    <input type="@(prevDisabled ? "hidden" : "button")" readonly="readonly" value="Previous"
+                           class="btn btn-primary" @onclick="@PreviousPage" />
+
+                    <input class="btn btn-outline-primary col-sm-1" value="@pageIndex" />
```
Comment (dTarasova) [inline Clients/GUI/Pages/Portfolio.razor]: have to make it readonly

File: Clients/GUI/Pages/Portfolio.razor [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+        if (firstRender)
+        {
+            try
+            {
+                areInstrumentsLoading = true;
+                userToken.UserId = await Storage.GetItemAsync<Guid>("id");
+                userToken.Body = await Storage.GetItemAsync<string>("token");
+                tinkoffToken = "t.-lW-yXwRofmPWDrSqm9XpBO2K4WbTmHPkKPgRskRDlMqLCxzbP8efGmM3l6U1HU9ReMg5R0t3QqO86twfG3-vw";
+
+                await Storage.SetItemAsync("tinkoffToken", tinkoffToken);
```
Comment (dTarasova) [inline Clients/GUI/Pages/Portfolio.razor:234]: Do we need to put it there?

File: Clients/GUI/wwwroot/css/mobilestyle.css [hunk 3]
```diff
@@ -0,0 +1,31 @@
+﻿@media only screen and (max-width: 760px), (min-device-width: 768px) and (max-device-width: 1024px) {
```
Comment (dTarasova) [inline Clients/GUI/wwwroot/css/mobilestyle.css:1]: Where is this style suppose to use

File: Services/DataBaseService/BrokerConsumers/GetPortfolioConsumer.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+namespace DataBaseService.BrokerConsumers
+{
+    public class GetPortfolioConsumer : IConsumer<GetPortfolioRequest>
+    {
+        private ITradeRepository repository;
+
+        public GetPortfolioConsumer([FromServices] ITradeRepository repository)
+        {
+            this.repository = repository;
```
Comment (dTarasova) [inline Services/DataBaseService/BrokerConsumers/GetPortfolioConsumer.cs:18]: logs?

File: Services/DataBaseService/Repositories/TradeRepository.cs [hunk 5]
```diff
@@ -95,7 +95,7 @@ private void UpdatePortfolioAfterTransaction(Transaction transaction)
             {
                 var exception = new BadRequestException("Not enough instrument count to sell");
                 logger.LogWarning(exception,
-                    $"User {transaction.UserId} asked to sell more instruments {transaction.Figi} than he has");
+                    $"{Guid.NewGuid()}_User {transaction.UserId} asked to sell more instruments {transaction.Figi} than he has");
```
Comment (dTarasova) [inline Services/DataBaseService/Repositories/TradeRepository.cs:98]: Why do we need this guid?

File: Services/OperationService/Commands/GetPortfolioCommand.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+        public async Task<List<InstrumentData>> Execute(GetPortfolioRequest request)
+        {
+            try
+            {
+                return await GetPortfolio(request);
+            }
+            catch
+            {
+                // tratata
```
Comment (dTarasova) [inline Services/OperationService/Commands/GetPortfolioCommand.cs]: :)

File: Services/OperationService/Commands/GetPortfolioCommand.cs [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+namespace OperationService.Commands
+{
+    public class GetPortfolioCommand : ICommand<GetPortfolioRequest, List<InstrumentData>>
+    {
+        private IRequestClient<GetPortfolioRequest> client;
+
+        public GetPortfolioCommand([FromServices] IRequestClient<GetPortfolioRequest> client)
+        {
+            this.client = client;
```
Comment (dTarasova) [inline Services/OperationService/Commands/GetPortfolioCommand.cs:24]: logs

File: TPlatform.sln [hunk 8]
```diff
@@ -61,7 +61,7 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "WebTests", "WebTests", "{62
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "GUITests", "GUITests\GUITests.csproj", "{84259840-2408-413F-9720-67BA90EE98A4}"
 EndProject
```
Comment (dTarasova) [inline TPlatform.sln:63]: Why do you need to change this?

**Review by dTarasova — COMMENTED** [review]
Fix colums size in the portfolio table, 
try to slow down changes in total portfolio price.

And instrument info page should be rest the same or swap graphic and information and make grafik bigger

**Review by VLiamin — APPROVED** [review]

File: Services/DataBaseService/Repositories/TradeRepository.cs (hunk 5 above)
Comment (nmamro) [reply Services/DataBaseService/Repositories/TradeRepository.cs:98]: > Why do we need this guid?

This is made for ParentId in logs

File: Clients/GUI/wwwroot/css/mobilestyle.css (hunk 3 above)
Comment (nmamro) [reply Clients/GUI/wwwroot/css/mobilestyle.css:1]: > Where is this style suppose to use

In mobile version of website

File: Services/OperationService/Commands/GetPortfolioCommand.cs (hunk 6 above)
Comment (nmamro) [reply Services/OperationService/Commands/GetPortfolioCommand.cs]: > :)

:D, will remove it

File: TPlatform.sln (hunk 8 above)
Comment (nmamro) [reply TPlatform.sln:63]: > Why do you need to change this?

No idea, that's some automatic changes

File: Clients/GUI/Pages/Portfolio.razor (hunk 2 above)
Comment (nmamro) [reply Clients/GUI/Pages/Portfolio.razor:234]: > Do we need to put it there?

What if user opens portfolio before instruments? Or im wrong?

File: Services/OperationService/Commands/GetPortfolioCommand.cs (hunk 7 above)
Comment (nmamro) [reply Services/OperationService/Commands/GetPortfolioCommand.cs:24]: > logs

Ok, but I really don't like logging like EVERY FREAKING THING, generating thousands of logs after using app just for a few hours, for example, and this only includes usage by a single person
🙃

**Review by Cerberous007 — APPROVED** [review]

**Review by ifelseelif — APPROVED** [review]

File: Clients/GUI/Pages/Portfolio.razor (hunk 2 above)
Comment (dTarasova) [reply Clients/GUI/Pages/Portfolio.razor:234]: Yeah, okay, but I thought we just put token everywhere and don't ask user to put it anywhere 

Why do we have to use storage if it is just const variable

File: Services/DataBaseService/Repositories/TradeRepository.cs (hunk 5 above)
Comment (dTarasova) [reply Services/DataBaseService/Repositories/TradeRepository.cs:98]: It is just the first line? We have another field for parent id. Why do we have to add id in the message?

File: TPlatform.sln (hunk 8 above)
Comment (dTarasova) [reply TPlatform.sln:63]: Probably have to discard this changes in otrder to not have conflicts with other feature

**Review by denis-kriv — APPROVED** [review]

**Pull request merged**